### PR TITLE
ref(js): Convert SetupWizard to a FC

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Also removes the only usage of `finished` from the LoadingIndicator

### CODE REVIEW

File: static/app/views/setupWizard/index.tsx [hunk 1]
```diff
@@ -1,79 +1,50 @@
...
-          this.setState({finished: true});
-          setTimeout(() => window.close(), 10000);
-        },
-      });
-    });
+function SetupWizard({hash = false}: Props) {
+  const api = useApi();
+  const [finished, setFinished] = useState(false);
+
+  async function checkFinished() {
```
Comment (evanpurkhiser) [inline static/app/views/setupWizard/index.tsx:16]: This is just checking if we can find the wizard hash.

File: static/app/views/setupWizard/index.tsx [hunk 2]
```diff
@@ -1,79 +1,50 @@
...
-      this.props.api.request(`/wizard/${this.props.hash}/`, {
-        method: 'GET',
-        success: () => {
-          setTimeout(() => this.pollFinished(), 1000);
-        },
-        error: () => {
-          resolve();
-          this.setState({finished: true});
-          setTimeout(() => window.close(), 10000);
-        },
```
Comment (evanpurkhiser) [inline static/app/views/setupWizard/index.tsx:41]: I don't really understand how this used to work 🤔

**Review by priscilawebdev — APPROVED** [review]
code looks good

**Comment by evanpurkhiser** [comment]
I didn't test this, so would be nice if I could convince @HazAT to test 😬

File: static/app/views/setupWizard/index.tsx (hunk 2 above)
Comment (priscilawebdev) [reply static/app/views/setupWizard/index.tsx:41]: Agree.. this is a little strange... so when the API returns an error..it means that the setup is done?

Comment (evanpurkhiser) [reply static/app/views/setupWizard/index.tsx:41]: https://github.com/getsentry/sentry/blob/80ecaaa5a31ffa379f0a0898010be20678b76d7a/src/sentry/api/endpoints/setup_wizard.py#L28-L56

The endpoint didn't offer any great clues either :grim

**Comment by evanpurkhiser** [comment]
verified this works locally

**Pull request merged**